src/script/signingprovider.h

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_SCRIPT_SIGNINGPROVIDER_H
+#define BITCOIN_SCRIPT_SIGNINGPROVIDER_H
+
+#include <key.h>
+#include <pubkey.h>
+#include <script/script.h>
+#include <script/standard.h>
+#include <sync.h>
+
+struct KeyOriginInfo;
+
+/** An interface to be implemented by keystores that support signing. */
+class SigningProvider
```

Comment:
Any thoughts on:

(1) turning `SigningProvider` into a "pure interface" that only declares pure virtual functions?
(2) isolating `SigningProvider` and define it in its own header file?

These changes could make sense because we are using `SigningProvider` as an interface.

Reply:
I think it is fine to leave it here. Making it a pure interface does make sense.